src/PrestaShopBundle/Service/Mail/MailTemplateFolderCatalog.php

```diff
@@ -0,0 +1,191 @@
+<?php
+/**
+ * 2007-2018 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Service\Mail;
```

Comment:
wow: why everything is stored in the PrestaShopBundle?

Use of Service namespace is "kind of" deprecated https://devdocs.prestashop.com/1.7/development/architecture/understanding-src-folder/#prestashopbundle

Reply:
I agree, I already moved all the classes in Core\MailTemplate
except the renderer which is now in Adapter\MailTemplate\MailTemplateTwigRenderer
(wait for the next push)